src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/AddConfigureAwaitAnalyzer.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Azure.ClientSdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class AddConfigureAwaitAnalyzer : DiagnosticAnalyzer
+    {
+        private AsyncAnalyzerUtilities _asyncUtilities;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(Descriptors.AZC0012, Descriptors.AZC0013);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.RegisterCompilationStartAction(CompilationStart);
+        }
+
+        private void CompilationStart(CompilationStartAnalysisContext context) 
+        {
+            _asyncUtilities = new AsyncAnalyzerUtilities(context.Compilation);
+            context.RegisterSyntaxNodeAction(AnalyzeAwaitExpression, SyntaxKind.AwaitExpression);
+            context.RegisterSyntaxNodeAction(AnalyzeUsingExpression, SyntaxKind.UsingStatement);
+            context.RegisterSyntaxNodeAction(AnalyzeForEachExpression, SyntaxKind.ForEachStatement);
+            context.RegisterSyntaxNodeAction(AnalyzeConfigureAwaitTrue, SyntaxKind.InvocationExpression);
+        }
+
+        private void AnalyzeAwaitExpression(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (AwaitExpressionSyntax)context.Node;
```

Comment:
Can this be something else?

Reply:
Below you didn't cast the operation -- not sure if there is a reason to cast here.

Reply:
Good catch, this cast isn't needed.